src/App.tsx

```diff
@@ -1,14 +1,98 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterType } from './types/Filter';
+import { filterActiveOrCompletedTodosBySearchQuery, filterTodosBySearchQuery } from './services/Filtration';
 
 export const App: React.FC = () => {
+  const [loading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<string>('all');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos().then((todosList) => {
+      setTodos(todosList);
+      setFilteredTodos(todosList);
+      setLoading(false);
+    });
+  }, []);
+
+  useEffect(() => {
+    let filtered;
+
+    if (searchQuery === '') {
+      switch (filter) {
+        case FilterType.ALL:
+          filtered = todos;
+          break;
+
+        case FilterType.COMPLETED:
+          filtered = todos.filter(todo => todo.completed);
+          break;
+
+        case FilterType.ACTIVE:
+          filtered = todos.filter(todo => !todo.completed);
+          break;
+
+        default:
+          filtered = todos;
+          break;
+      }
+    } else {
+      switch (filter) {
+        case FilterType.ALL:
+          filtered = filterTodosBySearchQuery(todos, searchQuery);
+          break;
+
+        case FilterType.COMPLETED:
+          filtered = filterActiveOrCompletedTodosBySearchQuery(todos, searchQuery, true);
+          break;
+
+        case FilterType.ACTIVE:
+          filtered = filterActiveOrCompletedTodosBySearchQuery(todos, searchQuery, false);
+          break;
+
```

Comment:
looks complicated, you can just filter the todos in two steps
1. filter by `filter` state
2. filter by `query`